Modules/selectmodule.c

```diff
@@ -532,11 +532,12 @@ poll_poll(pollObject *self, PyObject *args)
         return NULL;
     }
 
+    deadline = 0;   /* initialize to prevent gcc warning */
+
     /* Check values for timeout */
     if (timeout_obj == NULL || timeout_obj == Py_None) {
         timeout = -1;
         ms = -1;
-        deadline = 0;   /* initialize to prevent gcc warning */
     }
     else {
         if (_PyTime_FromMillisecondsObject(&timeout, timeout_obj,
```

Comment:
Another question: Can't we just set `timeout` to `INFTIM` or `-1` if `timeout_obj` is less than `-1` instead of messing with `ms`?

Reply:
First of all thanks for the review.

I'm not getting why this would be better. The current checks against `INT_MIN` and `INT_MAX` are done with `ms`, so I followed that pattern. Also `ms` is the actual parameter passed to the system call and the one causing the issue right now, it seemed more logic to me to change directly this one.